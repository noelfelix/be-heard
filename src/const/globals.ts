/* istanbul ignore file */

import { useTranslation as useTranslationi18next } from "react-i18next";
import { Environment } from "../env";
import { getFeatureFlag } from "../views/features/featureFlags";
import { FEATURE_FLAGS } from "../views/features/availableFeatures";

export const rootContext = "/app";
export const legacyActonContext = "/acton";

declare const __PROD__: boolean;
declare const __TEST__: boolean;
declare const __DEV__: boolean;
declare const __DEBUG__: boolean;
declare const __LOCAL_DEV__: boolean;
declare const __PATH__: string;
declare const jest: any;

export function isProd(): boolean {
  return __PROD__;
}

export function isTest(): boolean {
  return __TEST__;
}

export function isDev(): boolean {
  return __DEV__;
}

export function isDebug(): boolean {
  return __DEBUG__;
}

export function isLocal(): boolean {
  return __LOCAL_DEV__;
}

export function getPath(): string {
  return __PATH__;
}

export function getEnvironment(): Environment {
  return (window as any).__ENV__;
}

export function setEnvironment(env: Environment) {
  (window as any).__ENV__ = env;
}

export function isJest() {
  let isJest = false;
  try {
    isJest = jest !== undefined;
  } catch (e) {}
  return isJest;
}

export function useTranslation() {
  return isJest() ? { t: (s: any) => s } : useTranslationi18next();
}

export function isMocks() {
  return getFeatureFlag(FEATURE_FLAGS.MOCK) === true;
}

export default {
  isProd,
  isTest,
  isDev,
  isDebug,
  isLocal,
  getPath,
  getEnvironment,
  setEnvironment,
  useTranslation,
  isMocks,
};
