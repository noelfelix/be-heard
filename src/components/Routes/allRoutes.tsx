import { ComponentClass, FC } from 'react'
import Features from '../../views/features'
import { rootContext } from '../../const/globals'

export interface RouteConfig {
  path: string
  component: FC | ComponentClass | null
  noProd?: boolean
}

const allRoutes: RouteConfig[] = [
  {
    path: `${rootContext}/classic`,
    component: null,
  },
  {
    path: `${rootContext}/features`,
    component: Features,
    noProd: true,
  },
]

export default allRoutes
