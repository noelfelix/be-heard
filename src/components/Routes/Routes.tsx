import React, { FC } from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'

import gobals, { rootContext } from 'const/globals'
import allRoutes, { RouteConfig } from './allRoutes'

export const getRoutes = (routes: RouteConfig[]) =>
  routes.map((route) => {
    if (route.noProd && gobals.isProd()) return null

    if (route.component) {
      return <Route key={route.path} path={route.path} component={route.component} />
    }
    return <Route key={route.path} path={route.path} />
  })

const Routes: FC = () => {
  return (
    <Switch>
      {...getRoutes(allRoutes)}
      <Route path="/" exact={true}>
        <Redirect to={`${rootContext}/classic/`} />
      </Route>
      <Route path="" exact={true}>
        <Redirect to={`${rootContext}/classic/`} />
      </Route>
      <Route path="*">
        <Redirect to={`${rootContext}/classic/404`} />
      </Route>
    </Switch>
  )
}

export default Routes
