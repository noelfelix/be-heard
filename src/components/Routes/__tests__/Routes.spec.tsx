import React from 'react'
import { shallow } from 'enzyme'
import globals from 'const/globals'
import Routes from '../Routes'

describe('Routes', () => {
  it('render Routes default', () => {
    const wrapper = shallow(<Routes />)
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('render Routes for production', () => {
    jest.spyOn(globals, 'isProd').mockImplementation(() => true)
    const wrapper = shallow(<Routes />)
    expect(wrapper.debug()).toMatchSnapshot()
  })
})
