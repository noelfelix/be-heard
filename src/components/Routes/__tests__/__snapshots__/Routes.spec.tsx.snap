// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes render Routes default 1`] = `
"<Switch>
  <Route path=\\"/app/classic\\" />
  <Route path=\\"/app/features\\" component={[Function]} />
  <Route path=\\"/\\" exact={true}>
    <Redirect to=\\"/app/classic/\\" />
  </Route>
  <Route path=\\"\\" exact={true}>
    <Redirect to=\\"/app/classic/\\" />
  </Route>
  <Route path=\\"*\\">
    <Redirect to=\\"/app/classic/404\\" />
  </Route>
</Switch>"
`;

exports[`Routes render Routes for production 1`] = `
"<Switch>
  <Route path=\\"/app/classic\\" />
  <Route path=\\"/\\" exact={true}>
    <Redirect to=\\"/app/classic/\\" />
  </Route>
  <Route path=\\"\\" exact={true}>
    <Redirect to=\\"/app/classic/\\" />
  </Route>
  <Route path=\\"*\\">
    <Redirect to=\\"/app/classic/404\\" />
  </Route>
</Switch>"
`;
